Intro-Full/Exercises/06/Solution/exercise_6_solution.cpp

```diff
@@ -148,21 +148,21 @@ int main( int argc, char* argv[] )
     // Application: <y,Ax> = y^T*A*x
     double result = 0;
 
-    Kokkos::parallel_reduce( team_policy( E, Kokkos::AUTO, 32 ), KOKKOS_LAMBDA ( const member_type &teamMember, double &update ) {
+    Kokkos::parallel_reduce( "yAx_outer", team_policy( E, Kokkos::AUTO, 32 ), KOKKOS_LAMBDA ( const member_type &teamMember, double &update ) {
       const int e = teamMember.league_rank();
       double tempN = 0;
 
-      Kokkos::parallel_reduce( Kokkos::TeamThreadRange( teamMember, N ), [&] ( const int j, double &innerUpdateN ) {
+      Kokkos::parallel_reduce( "yAx_intermediate", Kokkos::TeamThreadRange( teamMember, N ), [&] ( const int j, double &innerUpdateN ) {
```

Comment:
I'm pretty sure it's either impossible or meaningless to label inner `parallel_*` or `single`.

Reply:
You mean it is probably inlined anyway? Did not check actually with hierarchical parallelism. Shall we just not name them?

Reply:
@ax3l wrote:
> You mean it is probably inlined anyway?

What I mean is that the string only has an effect for the outermost `parallel_*` function.  Labels  mainly exist for Kokkos profiling, and profiling only happens at the beginning and end of the outermost `parallel_*` function.  However, if this compiles, I'm pretty sure it's harmless to leave the labels in place -- Kokkos could in theory use them for debugging.

Reply:
I removed them anyway, according to @crtrott it has no effect so far and the names I gave were pretty generic anyway.